fibonacci.asm

```diff
@@ -17,15 +17,24 @@
 	.data
 
 fibs:	.word   0 : 9         # create an array variable named "fibs" of 9 word-length elements (4 bytes each)
-size:	.word  9              # create a single integer variable named "size" that indicates the length of the array
+#size:	.word  9              # create a single integer variable named "size" that indicates the length of the array
+input: 	.asciiz	"\Enter the size of the array\n"
 
 ###############################################
 # .text segment. Assembly instructions go here.
 ###############################################
 	.text
+	      la   $a0, input	#load input from user to a0
+	      li   $v0, 4	#load call 4 to display text	
+	      syscall		#do
+	      
+	      li   $v0, 5	#load inmmediate call 5
+	      syscall		#do	
+	      
+	      move $s5, $v0	#move and replace the value in s5 (where array size will be store) with v0 (array size from input user)			
```

Comment:
-5 Not limited to 9 fibonacci numbers.  Very dangerous to allow the user to exceed the bounds of the array set up in the beginning because you may overwrite other important addresses.  May or may not force the printing of unintended characters, depending on whether or not one of the fibonacci numbers writes over any of the print strings.